rfcs/0054-support-levels-and-tiers.md

```diff
@@ -0,0 +1,171 @@
+Having Hydra evaluations representing the scope of each support level (i. e. stdenv for `core` and `bootstrap`, NixOS for `system`) should allow detecting packages with insufficient support level.
+Note that the evaluations should be mostly independent of platforms, since this is only about automatically detecting that packages required for a given scope are actually marked with at least the corresponding support level; the derivations needn't even be built for this purpose.
+
+Should restructuring (e. g. of stdenv bootstrap or changing the 'reference' NixOS configuration for the `system` support level/tier) exclude a package from the scope of a certain support level, the `meta.supportlevel` should be widened appropriately. There is no obvious way to check whether this is done correctly by automatic means.
+
+## Management of support/acceptance tiers
+(Note: this part in particular is a first draft. Suggestions welcome)
+
+As long as the relevant compiler supports the target, adding a platform to satisfy support tier `core` should be relatively simple. (Mainly involves adding support in `lib/platforms` and adding flags to compiler and libc if necessary)
+For that reason, acceptance of such platforms should be at `core` by default.
+
+Platform maintainer groups (PMGs) should exist for each supported platform (group). A single platform tuple may be covered by the scope of multiple PMGs (if one e. g. maintains x86 platforms and one maintains musl platforms).
+
+For lower support tiers (`core`, `bootstrap`) a PMG may only consist of a few individuals that can be CCed on relevant issues/PRs. If maintainers become irresponsive over extended periods of time, they may be removed from the group.
+There shall be lower limits on the number of active maintainers in the PMG for each support level (Note that this requirement is mostly to guarantee responsiveness for important platforms, since most breakage is expected to be detected using CI):
+
+* `core`: 1 maintainer
+* `bootstrap`: 2 maintainers
+* `system`: 5 maintainers
+* `full`: 10 maintainers
+
+PMGs should be notified and should publically give an (informal) report on the state of the platform as part of the release process for a major nixpkgs/NixOS version.
+This can also serve as an occasion to gauge responsiveness of their individual members.
+
+If the maintainers report maturity at the current support tier, the acceptance tier may be raised one above the support tier.
+If then the PMG reports adequate quality of support for the next-highest support tier, the support tier may be raised appropriately.
+
+Support tiers are dropped if the number of responsive maintainers in the PMG goes below the required number for the support level or if major breakage is reported for multiple subsequent major releases. (Note that the acceptance tier may be kept if sensible)
```

Comment:
Is the suport-acceptance gap allowed to become _two_ in this case?

Reply:
Good question. I'd argue it shouldn't, since the purpose of having a small gap is making maintainers focus on the basics first by denying PRs for minor packages. In practice I can't imagine a platform mature enough to warrant raising the acceptance level suddenly being abandoned. I suppose it might make sense to prescribe revoking the elevated acceptance level before dropping the support level.

Reply:
@CrystalGamma I can _remember_ such a thing in Nixpkgs, actually. Depending on the details of the defnitions, `i686-linux` might have eventually dropped from the flagship `all`-tier platform to a platform where part of `system`-level packages might be broken without too many people noticing.

Reply:
`i686-linux` in particular might be gnarly because its main use nowadays is as a dependency for wine on `x86_64-linux`, so it doesn't need everything from `system` but does need a lot of things like sound libraries that are out of scope for `system` … we could get around that by defining support levels/tiers not as total orders, but lattices, with an additional `wine` support tier/level as a sibling of `system`, but that sounds like it could easily lead to exactly the kind of complexity I was trying to avoid :(

Reply:
You have an option to say that there is an `x86_64-linux` Wine package (with multiarch support) that still depends on the same set of things, so these are supported in the interest of `x86_64-linux` by Wine package maintainers.

I do hope that the reference system configuration for the `system` package includes _some_ kind of sound support, though!

But I think that we cannot exclude that some other platform goes the same way.

Reply:
Sure, `system` should probably include things like libpulse and whatever the low-level libraries for ALSA are called … I was thinking of libraries like OpenAL or cubeb (then again I don't actually know what wine uses, I was just recalling some dependencies of random video games I've seen run on Ubuntu and assumed wine might use those as well).

Reply:
`vkd3d` might be a good example of a thing useful for Wine and really not making sense in `system`

Reply:
I suppose it makes sense to say that `i686-linux` inherits `x86_64-linux`'s acceptance tier where it makes sense for multilib.
It's probably sensible to have a provision for multilib somewhere in the RFC, in case that, as you say, it becomes relevant for another platform.